scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -226,62 +256,151 @@ private static void askInteractively(Console console, UserInfo userInfo, Action
+
+            for (String user : sourceLoginNames) {
+                UserInfo sourceUserInfo = new UserInfo();
+                sourceUserInfo.setLogin(user);
+                sourceUserInfo.setPassword((String) sourceLoginProps.get(user));
+                if (sourceGroupsMap.containsKey(user)) {
+                    sourceUserInfo.setGroups(sourceGroupsMap.get(user));
+                }
+
+                switch (action) {
+                    case CREATE:
+                        createAccount(pubKey, sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap, bulkMode);
+                        break;
+                    case UPDATE:
+                        updateAccount(pubKey, sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap, bulkMode);
+                        break;
+                    case DELETE:
+                        deleteAccount(sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap);
+                        break;
+                }
             }
 
-            storeLoginFile(loginFilePath, props);
+            storeLoginFile(loginFilePath, destinationLoginProps);
+
+            storeGroups(groupFilePath, destinationGroupsMap);
 
-            storeGroups(groupFilePath, groupsMap);
         } catch (Throwable t) {
             exitWithErrorMessage("Unexpected error", null, t);
         }
     }
 
+    private static void deleteAccount(UserInfo userInfo, String loginFilePath, String groupFilePath, Properties props, Multimap<String, String> groupsMap) throws ManageUsersException {
+        if (userInfo.getLogin().isEmpty()) {
+            warnWithMessage("Provided username is empty, skipping.");
+            return;
+        }
+        if (!props.containsKey(userInfo.getLogin())) {
+            warnWithMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath);
+        }
+        if (!groupsMap.containsKey(userInfo.getLogin())) {
+            warnWithMessage("USER " + userInfo.getLogin() + " does not exist in group file : " + groupFilePath);
+        }
+        props.remove(userInfo.getLogin());
+        groupsMap.removeAll(userInfo.getLogin());
+        System.out.println("Deleted user " + userInfo.getLogin());
+    }
+
+    private static void updateAccount(PublicKey pubKey, UserInfo userInfo, String loginFilePath, String groupFilePath, Properties props, Multimap<String, String> groupsMap, boolean bulkMode) throws ManageUsersException, KeyException {
+        if (userInfo.getLogin().isEmpty()) {
+            warnWithMessage("Provided username is empty, skipping.");
+            return;
+        }
+        if (!props.containsKey(userInfo.getLogin())) {
+            if (userInfo.getPassword() != null && userInfo.getGroups() != null) {
+                warnWithMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath + ", create this user.");
```

Comment:
The sentence  _"USER " + userInfo.getLogin() + " does not exist in login file : "_ is repeated 3 times. Please extract in a variable.